upgrade_guide/README.md

```diff
@@ -1,5 +1,60 @@
 ## Upgrade guide
 
+### 4.17.0
+
+#### Java17 is now a supported platform
+
+With [JAVA-3042](https://datastax-oss.atlassian.net/browse/JAVA-3042) the driver is now tested against Java17 as
+well as the other LTS JVM releases Java8 and Java11.  Based on our testing the Java driver appears to function
+normally on Java17, but of course bugs are always a possibility.  Issues arising from the Java17 platform will be
+triaged and addressed just like any other issue.
+
+#### Updated API for vector search
+
+The 4.16.0 release introduced support for the CQL vector datatype (see
```

Comment:
Eventually yes, but for now the code isn't released in any Cassandra or DSE version.  It will be available in Astra but nowhere else yet.

So eventually yeah, we'll prolly need to add more detailed information regarding dependencies... but I don't think we're there yet.